File: test-suite/src/test/java/io/micronaut/docs/operator/OperatorFiltersTest.java
package io.micronaut.docs.operator;

import io.kubernetes.client.openapi.models.V1ConfigMap;
import io.kubernetes.client.openapi.models.V1ObjectMeta;
import org.junit.jupiter.api.Test;

import java.util.Map;

import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;

class OperatorFiltersTest {

    private static V1ConfigMap configMap(Map<String, String> annotations) {
        return new V1ConfigMap().metadata(new V1ObjectMeta().name("test").namespace("default").annotations(annotations));
    }

    @Test
    void testOnAddFilter() {
        assertTrue(new OnAddFilter().test(configMap(Map.of("io.micronaut.operator", "processed"))));
        assertFalse(new OnAddFilter().test(configMap(Map.of("other", "value"))));
        assertFalse(new OnAddFilter().test(configMap(null)));
    }

    @Test
    void testOnUpdateFilter() {
        assertTrue(new OnUpdateFilter().test(configMap(null), configMap(Map.of("io.micronaut.operator", "processed"))));
        assertFalse(new OnUpdateFilter().test(configMap(Map.of("io.micronaut.operator", "processed")), configMap(null)));
    }

    @Test
    void testOnDeleteFilter() {
        assertTrue(new OnDeleteFilter().test(configMap(Map.of("io.micronaut.operator", "processed")), false));
        assertFalse(new OnDeleteFilter().test(configMap(null), true));
    }
}

File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/operator/OperatorFiltersTest.kt
package io.micronaut.docs.operator

import io.kubernetes.client.openapi.models.V1ConfigMap
import io.kubernetes.client.openapi.models.V1ObjectMeta
import org.junit.jupiter.api.Assertions.assertFalse
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test

class OperatorFiltersTest {

    private fun configMap(annotations: Map<String, String>?) =
        V1ConfigMap().metadata(V1ObjectMeta().name("test").namespace("default").annotations(annotations?.toMutableMap()))

    @Test
    fun testOnAddFilter() {
        assertTrue(OnAddFilter().test(configMap(mapOf("io.micronaut.operator" to "processed"))))
        assertFalse(OnAddFilter().test(configMap(mapOf("other" to "value"))))
        assertFalse(OnAddFilter().test(configMap(null)))
    }

    @Test
    fun testOnUpdateFilter() {
        assertTrue(OnUpdateFilter().test(configMap(null), configMap(mapOf("io.micronaut.operator" to "processed"))))
        assertFalse(OnUpdateFilter().test(configMap(mapOf("io.micronaut.operator" to "processed")), configMap(null)))
    }

    @Test
    fun testOnDeleteFilter() {
        assertTrue(OnDeleteFilter().test(configMap(mapOf("io.micronaut.operator" to "processed")), false))
        assertFalse(OnDeleteFilter().test(configMap(null), true))
    }
}
